editors/js/threeDVE.js

```diff
@@ -0,0 +1,1329 @@
+              if (tdVE.id === tdVE.layoutMgr.focusPaneID) {
+                tdVE.showAllCurrentAnnotations();
+              }
+            }
+            tdVE.api.addEventListener('annotationFocus', function(index) {
+              var segIDs = tdVE.findSegIDByAnnotationIndex(index);
+              if (segIDs) {
+                $('.editContainer').trigger('updateselection',[tdVE.id, segIDs]);
+              }
+              tdVE.loadAnnotationPropertyView(index);
+            });
+            tdVE.api.addEventListener('click', function (info) {
+              if (tdVE.annotateMode) {
+                tdVE.api.getCameraLookAt(function(err, camera) {
+                  var segID = tdVE.getEdnVESelectedSegID();
+                  var sclID = tdVE.getEdnVESelectedSclID();
+                  if (segID && sclID) {
+                    tdVE.addSyllableAnnotation(segID, sclID, info.position3D, camera.position, camera.target);
+                  } else {
+                    alert('A segment must be selected from the edition VE');
+                  }
+                  tdVE.exitAnnotateMode();
+                });
+              }
+            });
+            tdVE.addEventHandlers();
+          });
+          tdVE.api.start();
+          tdVE.createStaticToolbar();
+        },
+        error: function onError() {
+          alert('Failed to load the 3D viewer');
+        }
+      });
+    } else {
+      alert('There is no 3D model available for this edition');
+    }
+  },
+
+  /**
+   * Create all annotations in the 3D model.
+   */
+  createAnnotations: function () {
+    if (this.annotationData.syllable.length > 0) {
+      this.createModelAnnotation('syllable', 0, true);
+    }
+  },
+
+  /**
+   * Create a single annotation in the 3D model.
+   *
+   * This method is used to iterate the annotation data to create the
+   * annotations. Once one annotation is created, it will continue to create
+   * the next until all the annotation data are consumed.
+   *
+   * @param {string} objLevel The annotation object level, which could be
+   *   'syllable', 'token' or 'compound'.
+   * @param {int} anoDataIndex The index of the annotation data item.
+   * @param {boolean} hidden Hide/Show the created annotation
+   */
+  createModelAnnotation: function (objLevel, anoDataIndex, hidden) {
+    var tdVE = this;
+    this.api.createAnnotationFromWorldPosition(
+        this.annotationData[objLevel][anoDataIndex].coords,
+        this.annotationData[objLevel][anoDataIndex].cameraPosition,
+        this.annotationData[objLevel][anoDataIndex].cameraTarget,
+        this.annotationData[objLevel][anoDataIndex].title,
+        this.annotationData[objLevel][anoDataIndex].text,
+        function(err, index) {
+          if(!err) {
+            tdVE.annotationData[objLevel][anoDataIndex].index = index;
+            if (hidden) {
+              tdVE.api.hideAnnotation(index);
+            }
+            if (anoDataIndex + 1 < tdVE.annotationData[objLevel].length) {
```

Comment:
Please comment on algorithm.  Unclear that all patterns are handled. User gestor is always at object level, so are annotations shown only if they exist at the object level or is there some sort of hierarchy?

Reply:
Will have a look.

Reply:
@yangli0516, did you have a look? Can you resolve this?

Reply:
Added more comments.